### DISCUSSION TITLE
How to fetch data from api inside next.js

### DISCUSSION THREAD

**Discussion by medsghiri** [vercel/next.js#19532] [Help]
Hi I'm an absolute beginner I used isomorphic-fetch in the past to fetch data from an external API but I have no idea how to fetch data from an API inside next.js

```
import React from 'react'
import fetch from 'isomorphic-fetch'
class Index extends React.Component {

    static async getInitialProps({req, res, query}){
        let stories;
        let page;
        page = Number(query.page) || 1;
        const response = await fetch(
            `https://node-hnapi.herokuapp.com/news?page=${page}`
            );
        stories = await response.json();
```
**How to fetch data from** `http://localhost:3000/api/data`
Thank you

**Comment by kachkaev** [comment]
You don’t need `'isomorphic-fetch'` or any other polyfill library because Next.js polyfills `fetch` for you.

See https://nextjs.org/docs/basic-features/data-fetching

Also:

> **Note:** You should not use fetch() to call an API route in your application. Instead, directly import the API route and call its function yourself. You may need to slightly refactor your code for this approach.
> Fetching from an external API is fine!
